anyio/_backends/_asyncio.py

```diff
@@ -150,7 +150,9 @@ def run(func: Callable[..., T_Retval], *args, debug: bool = False, use_uvloop: b
         asyncio.set_event_loop_policy(policy)
 
     exception = retval = None
-    native_run(wrapper(), debug=debug)
+    loop = asyncio.get_event_loop()
+    loop.set_debug(debug)
+    loop.run_until_complete(wrapper())
```

Comment:
We're not using `native_run()` because it closes the event loop and shuts down async generators, thus rendering us unable to continue their execution on a successive call to `anyio.run()`. I can add the same comment to the code.

Reply:
Please do. Otherwise, looks sane enough to merge. ;-)